src/Microsoft.Owin/Extensions/AppBuilderCookieExtensions.cs

```diff
@@ -0,0 +1,111 @@
+        /// <param name="manager">The cookie manager.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="app"/> is <see langword="null"/>.</exception>
+        public static void SetDefaultCookieManager(this IAppBuilder app, ICookieManager manager)
+        {
+            if (app is null)
+            {
+                throw new ArgumentNullException(nameof(app));
+            }
+
+            if (manager is null)
+            {
+                app.Properties.Remove("infrastructure.CookieManager");
+            }
+            else
+            {
+                app.Properties["infrastructure.CookieManager"] = manager;
+            }
+        }
+
+        /// <summary>
+        /// Sets the default chunking <see cref="ICookieManager"/> that will be used
+        /// by the OWIN middleware when no manager is explicitly set in their options.
+        /// </summary>
+        /// <remarks>
+        /// Depending on the host, a default <see cref="ICookieManager"/> instance
+        /// may already be registered in <see cref="IAppBuilder.Properties"/>
+        /// (e.g SystemWeb automatically registers a cookie manager that relies on
+        /// HttpCookieCollection instead of directly manipulating the HTTP headers).
+        /// </remarks>
+        /// <param name="app">The application builder.</param>
+        /// <param name="manager">The cookie manager.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="app"/> is <see langword="null"/>.</exception>
+        public static void SetDefaultChunkingCookieManager(this IAppBuilder app, ICookieManager manager)
+        {
+            if (app is null)
+            {
+                throw new ArgumentNullException(nameof(app));
+            }
+
+            if (manager is null)
+            {
+                app.Properties.Remove("infrastructure.ChunkingCookieManager");
+            }
+            else
+            {
+                app.Properties["infrastructure.ChunkingCookieManager"] = manager;
+            }
+        }
+
+        /// <summary>
+        /// Gets the default <see cref="ICookieManager"/> that will be used by
+        /// the OWIN middleware when no manager is explicitly set in their options.
+        /// </summary>
+        /// <param name="app">The application builder.</param>
+        /// <returns>The cookie manager or <see langword="null"/> if no default instance was registered.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="app"/> is <see langword="null"/>.</exception>
+        public static ICookieManager GetDefaultCookieManager(this IAppBuilder app)
+        {
+            if (app is null)
+            {
+                throw new ArgumentNullException(nameof(app));
+            }
+
+            return app.Properties.TryGetValue("infrastructure.CookieManager", out object value) &&
+                value is ICookieManager manager ? manager : null;
```

Comment:
An alternate design could be to return `new CookieManager()` instead of `null` and remove `?? new CookieManager()` from the middleware.

Reply:
I tested this approach in the latest commit and it helps simplify things a bit 😃